### PR TITLE
autotools: Detect GTK version to use automatically

### PR DESCRIPTION
**Pull request by b4n**
Revamp #524 (actually, just cherry-pick the Autotools part on top of HEAD)

Closes #524.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -59,19 +59,28 @@ AC_CHECK_FUNCS([ftruncate fgetpos fnmatch mkstemp strerror strstr])
...
+		[enable_gtk3=auto])
+
+gtk2_package=gtk+-2.0
+gtk2_min_version=2.24
+gtk3_package=gtk+-3.0
+gtk3_min_version=3.0
+
+PKG_CHECK_EXISTS([$gtk2_package >= $gtk2_min_version], [have_gtk2=yes], [have_gtk2=no])
+PKG_CHECK_EXISTS([$gtk3_package >= $gtk3_min_version], [have_gtk3=yes], [have_gtk3=no])
+AS_IF([test "x$enable_gtk3" = xyes || (! test "x$enable_gtk3" = xno &&
```
Comment (codebrainz) [inline configure.ac]: Shouldn't it be an error if `--disable-gtk3` was passed and only GTK+ 3 is available? Also, how is `--enable-gtk3=auto` (the default) handled?

**Comment by codebrainz** [comment]
If this is going to flip the default for all distros, it might be worth discussing on the mailing lists. I personally will keep using GTK+ 2 build since GTK+ 3 gives me too many issues, but I've no problem passing `--disable-gtk3` to the configure script.

File: configure.ac (hunk 1 above)
Comment (elextr) [inline configure.ac]: `auto` is in `! test "x$enable_gtk3" = xno`

Comment (codebrainz) [inline configure.ac]: I see. It's also in `test "x$enable_gtk3" != "xno"` which reads more normally (or does the `!` apply to the whole set of expressions in the parenthesis?).

Comment (elextr) [inline configure.ac]: To the first test only according to my knowledge, but that is widely acknowledged as limited, so you better check it :)

**Comment by codebrainz** [comment]
This should probably wait until after #1181 to avoid a barrage of bug reports about missing VTE support.

**Comment by kugel-** [comment]
@b4n @elextr @codebrainz what about this for 1.30?

**Comment by kugel-** [comment]
@b4n @elextr @codebrainz this should be merged.

**Comment by codebrainz** [comment]
As GTK+ 3 is horrible (mainly broken theming on non-GNOME distros and user-hostile built-in smart phone dialogs and widgets), I won't be merging this myself.

**Comment by elextr** [comment]
On sensible distros like Mint the GTK3 version is fine, have been using it for ages.  Since #1181 is not committed this could be too, it doesn't change the GTK2 support so distros that want to remain GTK2 can do so, but I am aware of some that already provide the GTK3 version and that hasn't caused a massive complaint flood and will probably work better on Wayland distros.

@b4n If I read it correctly it doesn't change the default, just select gtk3 automatically if its the ONLY GTK available?

**Comment by kugel-** [comment]
Am 9. September 2017 01:58:23 MESZ schrieb elextr <notifications@github.com>:
>On sensible distros like Mint the GTK3 version is fine, have been using
>it for ages.  Since #1181 is not committed this could be too, it
>doesn't change the GTK2 support so distros that want to remain GTK2 can
>do so, but I am aware of some that already provide the GTK3 version and
>that hasn't caused a massive complaint flood and will probably work
>better on Wayland distros.
>
>@b4n If I read it correctly it doesn't change the default, just select
>gtk3 automatically if its the ONLY GTK available?
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/geany/geany/pull/1182#issuecomment-328239451

It doesnt change the default, indeed

File: configure.ac (hunk 1 above)
Comment (b4n) [reply configure.ac]: > I see. It's also in `test "x$enable_gtk3" != "xno"` which reads more normally

Changed for clarity

**Pull request merged**